openapi/schemas/Configuration/Configuration.json

```diff
@@ -0,0 +1,180 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "version": "1.0.0",
+    "title": "Propel Configuration API",
+    "description": "Propel Configuration API",
+    "license": {
+      "name": "MIT",
+      "url": "https://spdx.org/licenses/MIT"
+    },
+    "contact": {
+      "name": "Propel Support",
+      "url": "https://propel.force.com/helpcenter/s/",
+      "email": "support@propelsoftware.com"
+    }
+  },
+  "servers": [
+    {
+      "url": "https://{endpoint}/services/apexrest/{namespace}",
+      "description": "Propel API Endpoint Base URL",
+      "variables": {
+        "endpoint": {
+          "default": "login.salesforce.com",
+          "description": "This is the Salesforce instance url"
+        },
+        "namespace": {
+          "default": "PDLM",
+          "description": "This value should default to PDLM for managed package environments. For other environments, provide the namespace to your dev org."
+        }
+      }
+    }
+  ],
+  "paths": {
+    "/configuration/": {
+      "post": {
+        "summary": "Upsert a new configuration. A configuration is a custom setting or metadata within the package, the former being keyed by the 'Name' field whilst the latter on the 'Label' field.",
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json": {
+              "schema": {
+                "type": "object",
+                "oneOf": [
+                  {
+                    "$ref": "#/components/schemas/GoogleDriveSetting"
+                  },
+                  {
+                    "$ref": "#/components/schemas/SharingRule"
+                  }
+                ]
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Successfully created new Configuration",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object",
+                  "oneOf": [
+                    {
+                      "$ref": "#/components/schemas/GoogleDriveSetting"
+                    },
+                    {
+                      "$ref": "#/components/schemas/SharingRule"
```

Comment:
yes, but i dont think `openapi` has a wildcard or something to that effect. Else, this means in this spec we will need to enumerate every custom setting/metadata.

Also the idea of it is that the schema changes because you can potentially create a new setting or metadata type tomorrow and the API will still work with it. I was fiddling around with this today but could not find a way to properly 
give it a wildcard.

![image](https://github.com/PropelPLM/OpenAPI/assets/32610179/a6f828e7-4444-4693-a10d-df9af2d5d7e3)

Reply:
Wait, so would you prefer if I enumerated the custom setting/metadata? @arizonasingh

Reply:
@clawyq No I don't think that's necessary. The way you have it is fine. In my opinion, it's more clear if it's generically named like "CustomMetadataType" vs "SharingRule" but that's up to you. If you rename, there's some other implications, (e.g. you would have to make it clear that `Share_With_Field__c` is an example field, only Label is required for all types). Again, up to you since the primary use case is SharingRule so it's fine, but internally we should be aware that the API can be used with any type (unless your apex code specifically limits it to just SharingRule).